### PR TITLE
Add composed variant cairo args

### PR DESCRIPTION
**Pull request by SantiagoPittella**
# Add `Composed` variant in `CairoArg` enum.

## Description

This variant is used when the argument of a function is a composition of pointers.

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/runners/cairo_runner.rs [hunk 1]
```diff
@@ -964,6 +965,7 @@ impl CairoRunner {
             .map(|arg| vm.segments.gen_cairo_arg(arg, &mut vm.memory))
             .collect::<Result<Vec<MaybeRelocatable>, VirtualMachineError>>()?;
         let return_fp = vm.segments.add(&mut vm.memory);
+
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:968]: Please remove these added empty lines

File: src/vm/vm_core.rs [hunk 2]
```diff
@@ -334,8 +334,12 @@ impl VirtualMachine {
...
             if builtin.base() == address.segment_index {
                 match builtin.deduce_memory_cell(address, &self.memory) {
-                    Ok(maybe_reloc) => return Ok(maybe_reloc),
-                    Err(error) => return Err(VirtualMachineError::RunnerError(error)),
+                    Ok(maybe_reloc) => {
+                        return Ok(maybe_reloc);
+                    }
+                    Err(error) => {
+                        return Err(VirtualMachineError::RunnerError(error));
+                    }
```
Comment (fmoletta) [inline src/vm/vm_core.rs:342]: Please revert this change

**Review by juanbono — APPROVED** [review]

**Review by fmoletta — COMMENTED** [review]
Please add a simple tests for the added case to `gen_cairo_arg`

**Review by fmoletta — APPROVED** [review]
